src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -135,7 +135,7 @@ public async Task<SyncTriggersResult> TrySyncTriggersAsync(bool isBackgroundSync
 
                 bool shouldSyncTriggers = true;
                 string newHash = null;
-                if (isBackgroundSync && !_environment.IsKubernetesManagedHosting())
+                if (isBackgroundSync && !_environment.IsKubernetesManagedHosting() && hashBlobClient != null)
```

Comment:
Seems like you should also update this logic to `if (isBackgroundSync && hashBlobClient != null && !_environment.IsAnyKubernetesEnvironment())`, the goal being to skip the hash check across the board for Kubernetes, even if we DO have a blob client?